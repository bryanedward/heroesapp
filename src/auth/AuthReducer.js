import { types } from "../types/types";

const loginAuth = {
    types: types.login,
    payload: {
        name:'edward',
        password: '123'
    }
}

export const AuthReducer = ( state ={}, action) => {
    switch (action.type) {
        case types.login:
            return {
                ...action.payload,
                logged: true
            }            
            break;
        case types.logout:
            return {
                logged: false
            }
            break
    
        default:
            return state;
    }
}