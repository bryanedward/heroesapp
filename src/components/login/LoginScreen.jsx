import React, { useContext, useReducer } from 'react'
import { useNavigate } from 'react-router-dom'
import { AuthContext } from '../../auth/AuthContext'
import { AuthReducer } from '../../auth/AuthReducer'
import { types } from '../../types/types'

export const LoginScreen = () => {

  const navigate = useNavigate()

  const {dispatch} = useContext(AuthContext)

  const BtnAcceder = () => {
    const action = {
      type: types.login,
      payload: {
        name:'edward'
      }
    }
    dispatch(action)
    navigate('/marvel',{
      replace: true
    })
  }

   
  
  return (
    <div>
      <h4>login</h4>
      <button onClick={BtnAcceder}>
        Guardar
      </button>
    </div>
  )
}
