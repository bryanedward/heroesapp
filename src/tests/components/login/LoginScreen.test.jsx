import {mount} from 'enzyme'
import { MemoryRouter, Route, Routes } from 'react-router-dom'
import { AuthContext } from '../../../auth/AuthContext'
import { LoginScreen } from '../../../components/login/LoginScreen'
import { types } from '../../../types/types'


const mockNavigate = jest.fn()

jest.mock('react-router-dom',() => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate:() => mockNavigate
}))

const contextValue = {
    dispatch: jest.fn(),
    state:{
        name: 'edward'
    }
}
const wrapper = mount(<AuthContext.Provider value={contextValue}>
    <MemoryRouter initialEntries={['/login']}>
        <Routes>
            <Route path='/login' element={<LoginScreen/>}/>
        </Routes>
    </MemoryRouter>
</AuthContext.Provider>)


describe('componente del login', () => {
    test('validar el componente login', () => { 
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('h4').text().trim()).toBe('login')
     })
     test('validar el login del button', () => { 
        wrapper.find('button').prop('onClick')()
        expect(contextValue.dispatch).toHaveBeenCalledWith({
            type: types.login,
            payload: {name:'edward'}
        })
        expect(mockNavigate).toHaveBeenCalledWith("/marvel", {"replace": true})
      })
})