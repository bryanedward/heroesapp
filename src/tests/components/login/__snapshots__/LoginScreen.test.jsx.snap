// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`componente del login validar el componente login 1`] = `
<div>
  <h4>
    login
  </h4>
  <button
    onClick={[Function]}
  >
    Guardar
  </button>
</div>
`;
